From d0c0c7718f55514d83dd55760016e821adf558f1 Mon Sep 17 00:00:00 2001
From: Juha Vuolle <juha.vuolle@insta.fi>
Date: Mon, 7 Feb 2022 16:58:02 +0200
Subject: [PATCH 3/5] Keep the bluetooth service name if the remote has
 provided it

For example the "Bt Chat Server" example application service
would always show up as a "Custom Service".

Pick-to: 5.15 6.2 6.3
Change-Id: If73423d90d5681ee0c3655ef7ee3ad6cd7638755
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit e27c6600eca2c9b704c73a0b6d28780eb236cefb)
---
 src/bluetooth/qbluetoothservicediscoveryagent_bluez.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/bluetooth/qbluetoothservicediscoveryagent_bluez.cpp b/src/bluetooth/qbluetoothservicediscoveryagent_bluez.cpp
index 6d190711..4a18cfc7 100644
--- a/src/bluetooth/qbluetoothservicediscoveryagent_bluez.cpp
+++ b/src/bluetooth/qbluetoothservicediscoveryagent_bluez.cpp
@@ -320,7 +320,10 @@ void QBluetoothServiceDiscoveryAgentPrivate::_q_finishSdpScan(QBluetoothServiceD
             for (const QBluetoothUuid &id : serviceClassUuids) {
                 if (id.minimumSize() == 16) {
                     serviceInfo.setServiceUuid(id);
-                    serviceInfo.setServiceName(QBluetoothServiceDiscoveryAgent::tr("Custom Service"));
+                    if (serviceInfo.serviceName().isEmpty()) {
+                        serviceInfo.setServiceName(
+                                    QBluetoothServiceDiscoveryAgent::tr("Custom Service"));
+                    }
                     QBluetoothServiceInfo::Sequence modSeq =
                             serviceInfo.attribute(QBluetoothServiceInfo::ServiceClassIds).value<QBluetoothServiceInfo::Sequence>();
                     modSeq.removeOne(QVariant::fromValue(id));
-- 
2.36.1

